lib/connect/api/tierConfigRequests.js

```diff
@@ -0,0 +1,192 @@
+   * Some fields accept an array of values (search with the `in` operator).
+   *
+   * @param  {string|Query} filter                               A filter object as described below.
+   *
+   * @param  {string|Array} filter.type                            The request type (setup|update).
+   * @param  {string|Array} filter.status                          The status of the request (tiers_setup|pending|inquiring|approved|failed).
+   * @param  {string|Array} filter.id                              The id of the tier configuration request.
+   * @param  {string|Array} filter.configurationId                 The id of the tier configuration object that belongs to the request.
+   * @param  {string|Array} filter.configurationTierLevel          Tier level for product from customer perspective (1|2).
+   * @param  {string|Array} filter.configurationAccountId          The id of the tier account of the tier configuration that belongs to the request.
+   * @param  {string|Array} filter.configurationProductId          The id of the product of the tier configuration that belongs to the request.
+   * @param  {string|Array} filter.configurationAccountExternalUID The id of the user assignee of this tier configuration request.
+   * @param  {string}       filter.assigneeId                      The id of the user assignee of this tier configuration request.
+   * @param  {boolean}      filter.unassigned                      True to filter for unassigned requests, false otherwise.
+   * @param  {string|Array} orderBy                                The ordering criteria.
+   * @param  {number} [limit=100]                                  The maximum number of records to return.
+   * @param  {number} [offset=0]                                   The offset from which to start to include records.
+   * @return {Array} An array of Request objects that match the provided filters.
+   */
+  async list({
+    type,
+    status,
+    id,
+    configurationId,
+    configurationTierLevel,
+    configurationAccountId,
+    configurationProductId,
+    configurationAccountExternalUID,
+    assigneeId,
+    unassigned,
+  } = {}, orderBy, limit = 100, offset = 0) { /* eslint-enable max-len */
+    const params = { limit, offset };
+
+    addFilterParam(params, 'type', type);
+    addFilterParam(params, 'status', status);
+    addFilterParam(params, 'id', id);
+    addFilterParam(params, 'configuration__id', configurationId);
+    addFilterParam(params, 'configuration__tier_level', configurationTierLevel);
+    addFilterParam(params, 'configuration__account__id', configurationAccountId);
+    addFilterParam(params, 'configuration__product__id', configurationProductId);
+    addFilterParam(params, 'configuration__account__external_uid', configurationAccountExternalUID);
+    addFilterParam(params, 'assignee__id', assigneeId);
+    addFilterParam(params, 'unassigned', unassigned);
```

Comment:
Just a thought - maybe we could make something to add bulk params?
F.e. 'mergeDeepRight' from ramda or lodash 'assign'? 
will accept params, and k-v to assign

Reply:
It will change as soon as the endpoint exposes rql support.